lib/game/game.dart

```diff
@@ -23,6 +24,9 @@ abstract class Game {
   final assets = AssetsCache();
   BuildContext buildContext;
 
+  /// The list of component that is related to the game
+  final Iterable<Component> components = const Iterable.empty();
```

Comment:
Is this change correct? Only basegame has support for components right?

Reply:
It is due to the line on the other place where you commented on, since `HasGameRef` looks like this `HasGameRef<T extends Game>`. But maybe there is a better way to know whether `onMount` has already been run on a component?

Reply:
Or maybe we should add `bool isMounted`, that might be useful for other things too?

Reply:
Seems like a good idea, I think the `bool isMounted` could be better than to have this list here.

Or maybe, we should change the mixin? It could be `HasGameRef<T extends BaseGame>`?

Reply:
Updated to `isMounted`